### PR TITLE
WebKit deprecated request in favour of enter

### PR DESCRIPTION
**Pull request by mihar-22**
WebKit has deprecated `webkitRequestFullscreen` in favour of `webkitEnterFullscreen`.

Ref: https://github.com/sindresorhus/screenfull.js/issues/139

https://developer.apple.com/documentation/webkitjs/htmlvideoelement/1630649-webkitenterfullscreen?language=javascript

https://developer.apple.com/documentation/webkitjs/htmlvideoelement/1633500-webkitenterfullscreen?language=javascript

### CODE REVIEW

File: src/screenfull.js [hunk 1]
```diff
@@ -28,7 +28,7 @@
 			],
 			// Old WebKit
 			[
-				'webkitRequestFullScreen',
```
Comment (RusinovAnton) [inline src/screenfull.js]: Maybe it makes sense to leave the old one for older browsers to happen?

File: dist/screenfull.js [hunk 2]
```diff
@@ -23,7 +23,7 @@
 			],
 			// New WebKit
 			[
-				'webkitRequestFullscreen',
+				'webkitEnterFullscreen',
```
Comment (RusinovAnton) [inline dist/screenfull.js]: This won't work as the `webkitEnterFullscreen` only present in the HTMLVideoElement object, not in the `document`.  There should be a way to check whether video object has ability to enter fullscreen.

Comment (mihar-22) [reply dist/screenfull.js]: Ye you're right @RusinovAnton, not sure how I missed that. It seems like you can't request fullscreen on the document in Safari? I just tried and nothing worked. Has it always been like this?

Comment (RusinovAnton) [reply dist/screenfull.js]: I couldn't find actual info on that. I can see that there is no `webkitEnterFullscreen` property on `document` object in Safari. However. there is a `document.documentElement.webkitRequestFullscreen` which seems like not working for me. I've tried to find maybe there are some feature flags I can enable, but couldn't find any.

image

Comment (RusinovAnton) [reply dist/screenfull.js]: Uh-oh, it actually works if its triggered not from the console but from the user input even, like button click! - 

`button.addEventListener('click', ()=> document.documentElement.webkitRequestFullscreen())`

Comment (RusinovAnton) [reply dist/screenfull.js]: There is no `document.documentElement.webkitEnterFullscreen` property, tho

and the doc on it is pretty much empty :(

https://developer.apple.com/documentation/tvmljs/document/1630671-documentelement

**Pull request closed without merging**